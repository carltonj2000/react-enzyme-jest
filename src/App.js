import React, { Component } from "react";
import { Provider } from "react-redux";
import logo from "./logo.svg";
import "./App.css";
import TodoList from "./TodoList";
import { store } from "./store";
import Form from "./Form";

const Test = () => <div>Testing</div>;
const Title = ({ text }) => <div>{text}</div>;

class App extends Component {
  state = {
    button: "No!",
    title: "Start",
    mainColor: "blue",
    lifeCycle: "unmounted"
  };
  componentWillReceiveProps() {
    this.setState({ lifeCycle: "componentWillReceiveProps" });
  }
  componentDidMount() {
    this.setState({ lifeCycle: "componentDidMount" });
  }
  handleStrings(str) {
    return !!str;
  }
  render() {
    return (
      <Provider store={store}>
        <div className="App">
          <header className="App-header">
            <img src={logo} className="App-logo" alt="logo" />
            <h1 className="App-title">Welcome to React</h1>
          </header>
          <p className="App-intro">
            To get started, edit <code>src/App.js</code> and save to reload.
          </p>
          <Title text="hi cj" />
          <ul className="Carlton">
            <li>One</li>
            <li>Two</li>
            <li>Three</li>
          </ul>
          <Test />
          <button onClick={() => this.setState({ button: "Yes!" })}>
            {this.state.button}
          </button>
          <h2 className="title">{this.state.title}</h2>
          <input
            onChange={e => this.setState({ title: e.currentTarget.value })}
          />
          <div className={this.state.mainColor}>hi</div>
          <div className="lifeCycle">{this.state.lifeCycle}</div>
          <TodoList />
          <Form />
        </div>
      </Provider>
    );
  }
}

export class Link extends Component {
  render() {
    return this.props.hide ? null : <a href={this.props.address}>Click</a>;
  }
}

export default App;
