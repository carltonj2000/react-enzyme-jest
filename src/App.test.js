import React from "react";
import ReactDOM from "react-dom";
import App, { Link } from "./App";
import { configure, shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });

describe("<App /> shallow rendering", () => {
  it("should contain 1 p element", () => {
    const wrapper = shallow(<App />, {
      context: {},
      disableLifecycleMethods: true
    });
    //console.log(wrapper.debug());
    expect(wrapper.find("p").length).toBe(1);
    expect(wrapper.find(".App-intro").exists()).toBe(true);
    expect(wrapper.find("ul").children().length).toBe(3);
    expect(wrapper.find("ul").hasClass("Carlton")).toBe(true);
    expect(wrapper.find("h1").text()).toBe("Welcome to React");

    expect(wrapper.find('[text="hi cj"]').text()).toBe("<Title />");
    expect(wrapper.find({ alt: "logo" }).text()).toBe("");
  });
  it("matches the snapshot", () => {
    const tree = shallow(<App />);
    expect(toJson(tree)).toMatchSnapshot();
  });
  it("on button click changes text", () => {
    const wrapper = shallow(<App />);
    const button = wrapper.find("button");
    expect(button.text()).toBe("No!");
    button.simulate("click");
    expect(wrapper.find("button").text()).toBe("Yes!");
    //expect(button.text()).toBe("Yes!"); // this does not work
  });
  it("on input title changes text", () => {
    const wrapper = shallow(<App />);
    const input = wrapper.find("input");
    expect(wrapper.find(".title").text()).toBe("Start");
    input.simulate("change", { currentTarget: { value: "hi" } });
    expect(wrapper.find(".title").text()).toBe("hi");
  });
  it("updates className with new state", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find(".blue").length).toBe(1);
    expect(wrapper.find(".red").length).toBe(0);
    wrapper.setState({ mainColor: "red" });
    expect(wrapper.find(".blue").length).toBe(0);
    expect(wrapper.find(".red").length).toBe(1);
  });
  it("calls componentDidMount, update div text", () => {
    jest.spyOn(App.prototype, "componentDidMount");
    const wrapper = shallow(<App />);
    expect(App.prototype.componentDidMount.mock.calls.length).toBe(1);
    expect(wrapper.find(".lifeCycle").text()).toBe("componentDidMount");
  });
  it("setProps and componentWillReceiveProps", () => {
    jest.spyOn(App.prototype, "componentWillReceiveProps");
    const wrapper = shallow(<App />);
    wrapper.setProps({ hi: "there" });
    expect(App.prototype.componentWillReceiveProps.mock.calls.length).toBe(1);
    expect(wrapper.find(".lifeCycle").text()).toBe("componentWillReceiveProps");
  });
  it("handleStrings function returns correctly", () => {
    const wrapper = shallow(<App />);
    const trueReturn = wrapper.instance().handleStrings("hi");
    expect(trueReturn).toBe(true);
    const falseReturn = wrapper.instance().handleStrings("");
    expect(falseReturn).toBe(false);
  });
});

describe("<Link />", () => {
  it("Link component accepts address props", () => {
    const wrapper = shallow(<Link address="here" />);
    expect(wrapper.instance().props.address).toBe("here");
  });
  it("a tag node renders href correclty", () => {
    const wrapper = shallow(<Link address="here" />);
    expect(wrapper.props().href).toBe("here");
  });
  it("renders according to hide", () => {
    const wrapper = shallow(<Link address="here" hide={false} />);
    expect(wrapper.find("a").length).toBe(1);
    wrapper.setProps({ hide: true });
    expect(wrapper.get(0)).toBeNull();
  });
});

describe("<App /> mount rendering", () => {
  it("h1 has correct test", () => {
    const wrapper = mount(<App />);
    expect(wrapper.find("h1").text()).toBe("Welcome to React");
    wrapper.unmount();
  });
  it("matches the snapshot", () => {
    const tree = mount(<App />);
    expect(toJson(tree)).toMatchSnapshot();
    tree.unmount();
  });
});
