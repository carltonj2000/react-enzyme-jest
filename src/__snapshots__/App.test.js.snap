// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> mount rendering matches the snapshot 1`] = `
<App>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <div
      className="App"
    >
      <header
        className="App-header"
      >
        <img
          alt="logo"
          className="App-logo"
          src="logo.svg"
        />
        <h1
          className="App-title"
        >
          Welcome to React
        </h1>
      </header>
      <p
        className="App-intro"
      >
        To get started, edit 
        <code>
          src/App.js
        </code>
         and save to reload.
      </p>
      <Title
        text="hi cj"
      >
        <div>
          hi cj
        </div>
      </Title>
      <ul
        className="Carlton"
      >
        <li>
          One
        </li>
        <li>
          Two
        </li>
        <li>
          Three
        </li>
      </ul>
      <Test>
        <div>
          Testing
        </div>
      </Test>
      <button
        onClick={[Function]}
      >
        No!
      </button>
      <h2
        className="title"
      >
        Start
      </h2>
      <input
        onChange={[Function]}
      />
      <div
        className="blue"
      >
        hi
      </div>
      <div
        className="lifeCycle"
      >
        componentDidMount
      </div>
      <Connect(TodoList)>
        <TodoList
          addTodo={[Function]}
          removeTodo={[Function]}
          todos={Array []}
        >
          <ul />
          <input
            onChange={[Function]}
            value=""
          />
          <button
            onClick={[Function]}
          >
            Add Todo
          </button>
        </TodoList>
      </Connect(TodoList)>
      <Form>
        <form
          data-testid="addUserForm"
          onSubmit={[Function]}
        >
          <input
            data-testid="name"
            onChange={[Function]}
            value=""
          />
          <input
            data-testid="email"
            onChange={[Function]}
            value=""
          />
          <input
            data-testid="phone"
            onChange={[Function]}
            value=""
          />
          <input
            checked={true}
            data-testid="optIn"
            onChange={[Function]}
            type="checkbox"
          />
          <button
            data-testid="submitButton"
            onClick={[Function]}
          >
            Submit
          </button>
        </form>
      </Form>
    </div>
  </Provider>
</App>
`;

exports[`<App /> shallow rendering matches the snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <div
    className="App"
  >
    <header
      className="App-header"
    >
      <img
        alt="logo"
        className="App-logo"
        src="logo.svg"
      />
      <h1
        className="App-title"
      >
        Welcome to React
      </h1>
    </header>
    <p
      className="App-intro"
    >
      To get started, edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <Title
      text="hi cj"
    />
    <ul
      className="Carlton"
    >
      <li>
        One
      </li>
      <li>
        Two
      </li>
      <li>
        Three
      </li>
    </ul>
    <Test />
    <button
      onClick={[Function]}
    >
      No!
    </button>
    <h2
      className="title"
    >
      Start
    </h2>
    <input
      onChange={[Function]}
    />
    <div
      className="blue"
    >
      hi
    </div>
    <div
      className="lifeCycle"
    >
      componentDidMount
    </div>
    <Connect(TodoList) />
    <Form />
  </div>
</Provider>
`;
